join.spec.md

```diff
@@ -100,204 +101,347 @@ flowchart TB
+{ fieldA }
+## On B:
+{ fieldB }
 ```
 
-Documents MUST define a {join__Graph} enum. Each enum value defines a subgraph.
 
-:::[example](photos.graphql#join__Graph) -- Using join__Graph to define endpoints
+## Fields on the same subgraph as the parent operation
 
-The {join__Graph} enum is used as input to {@join} directives which link fields and types to subgraphs.
+If this field's parent field will be resolved by an operation on the same subgraph, then this field can be resolved as part of the same operation, by putting it in a nested selection set on the parent field's subgraph operation. Note that this example contains a {@join__owner} directive on an object type; this will be described later.
 
-# Directives
+```graphql example -- Fields on the same subgraph as the parent operation
+# Supergraph schema
+type Query {
+  fieldA: X @join__field(graph: A)
+}
 
-##! @join__graph
+type X @join__owner(graph: A) {
+  nestedFieldA: String @join__field(graph: A)
+}
 
-Declare subgraph metadata on {join__Graph} enum values. 
+# Operation
+{ fieldA { nestedFieldA } }
+# Generated subgraph operations
+## On A:
+{ fieldA { nestedFieldA }}
+```
 
-```graphql definition
-directive @join__graph(name: String!, url: String!) on ENUM_VALUE
+## Fields provided by the parent field
+
+When resolving a field to a composite type (object, interface, or union), it is sometimes easy to calculate the values of nested fields inside that object even if those nested fields are ordinarily resolved in a different subgraph. When this is the case, you can include a `provides` argument in the `@join__field` listing these "pre-calculated" fields. The router can now resolve these fields in the "providing" subgraph instead of in the subgraph that would usually be used to resolve those fields.
+
+```graphql example -- Provided fields
+# Supergraph schema
+type Query {
+  fieldA: X @join__field(graph: A, provides: "usuallyBField")
+  fieldB: X @join__field(graph: B)
+}
+
+type X @join__owner(graph: B) {
+  usuallyBField: String @join__field(graph: B)
+}
+
+# Operation
+{ fieldB { usuallyBField } }
+# Generated subgraph operations
+## On B
+{ fieldB { usuallyBField } }
+
+# Operation
+{ fieldA { usuallyBField } }
+# Generated subgraph operations
+## On A
+{ fieldA { usuallyBField } }
 ```
 
-:::[example](photos.graphql#join__Graph) -- Using {@join__graph} to declare subgraph metadata on the {join__Graph} enum values.
+## Fields on value types
 
-##! @join__type
+Some types have the property that all of their fields can be resolved by *any* subgraph that can resolve a field returning that type. These types are called *value types*. (Imagine a type `type T { x: Int, y: String }` where every resolver for a field of type `T` actually produces an object like `{x: 1, y: "z"}`, and the resolvers for the two fields on `T` just unpack the values already in the object.) In a supergraph schema, a type is a value type if it does not have a [{@join__owner}](#@join__owner) directive on it.
```

Comment:
Do we want to collapse "supergraph schema" to just "supergraph" everywhere?

Reply:
I think all the places I'm saying "supergraph" are talking about the actual schema document (whereas "supergraph" can also mean the server, the abstract concept of the data graph, etc) so I think it's an OK distinction?

Reply:
Aren't we talking about the document in this case (if we're talking about where directive usages are applied, etc.)

Reply:
Oops, typo in my last comment: I meant to say 

> all the places I'm saying "supergraph schema" are talking about the actual schema document

Does that make more sense?